apps/prairielearn/src/ee/lib/validateHTML.ts

```diff
@@ -5,6 +5,7 @@ type ChildNode = parse5.DefaultTreeAdapterMap['childNode'];
 
 const mustacheTemplateRegex = /^\{\{.*\}\}$/;
 const mustacheTemplateExtractorRegex = /\{\{((?:[^}]|\}[^}])*)\}\}/g;
+const mustacheTemplateChecker = /^\s*(((#|\/|\^|>|!|\$|>\*)?\s*(\w|\d|\.|_)*)|!.*)\s*$/;
```

Comment:
Rather than using regexes here, can we use a similar approach to #11962?

Reply:
@mwest1066 ha, that's precisely why I opened that PR first, I'm going to rebuild this one on top of that. The implementation should get much simpler.

Reply:
I added tests in f99e574.

Reply:
Very nice!